gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -106,22 +106,22 @@ public class Orchestrator implements SpecCatalogListener, Instrumentable {
 
   private UserQuotaManager quotaManager;
   private final FlowCompilationValidationHelper flowCompilationValidationHelper;
-  private Optional<FlowTriggerHandler> flowTriggerHandler;
+  private Optional<ReminderSettingFlowTriggerLeaseArbiter> flowTriggerDecorator;
   private Optional<FlowCatalog> flowCatalog;
   @Getter
   private final SharedFlowMetricsSingleton sharedFlowMetricsSingleton;
 
   @Inject
   public Orchestrator(Config config, TopologyCatalog topologyCatalog, DagManager dagManager,
-      Optional<Logger> log, FlowStatusGenerator flowStatusGenerator, Optional<FlowTriggerHandler> flowTriggerHandler,
+      Optional<Logger> log, FlowStatusGenerator flowStatusGenerator, Optional<ReminderSettingFlowTriggerLeaseArbiter> flowTriggerDecorator,
```

Comment:
the fact that we can't indicate the type here as `MultiActiveLeaseArbiter` is another clue that this type is not at heart a Decorator, but instead something that must `handleTriggerEvent`